### DISCUSSION TITLE
AttributeError: 'EmbeddingDatabase' object has no attribute 'skipped_embeddings'

### DISCUSSION THREAD

**Discussion by Misterkiwi500** [DominikDoom/a1111-sd-webui-tagcomplete#98] [General]
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/content/stable-diffusion-webui/extensions/a1111-sd-webui-tagcomplete/scripts/tag_autocomplete_helper.py", line 92, in get_embeddings
    while len(sd_hijack.model_hijack.embedding_db.word_embeddings) + len(sd_hijack.model_hijack.embedding_db.skipped_embeddings) < len(embs_in_dir):
AttributeError: 'EmbeddingDatabase' object has no attribute 'skipped_embeddings'

That's what occurred because of the error that probably shows me. I also like to use with google colab but can you fix that?

**Comment by DominikDoom** [comment]
This is due to an update to the webui shortly after I released the new embedding search system, will fix it soon.

**Reply by DominikDoom** [reply]
Actually seemed to work either way, the update today was only providing more info.
Just make sure your webui is up-to-date. The skipped_embeddings property was only added yesterday.

**Reply by Misterkiwi500** [reply]
ok, then I try

**Comment by Misterkiwi500** [comment]
![url](https://i.kym-cdn.com/photos/images/original/000/077/895/nope-3976_preview.png)

that's an error again!

Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/content/stable-diffusion-webui/extensions/a1111-sd-webui-tagcomplete/scripts/tag_autocomplete_helper.py", line 92, in get_embeddings
    while len(sd_hijack.model_hijack.embedding_db.word_embeddings) + len(sd_hijack.model_hijack.embedding_db.skipped_embeddings) < len(embs_in_dir):
AttributeError: 'EmbeddingDatabase' object has no attribute 'skipped_embeddings'

![error](https://user-images.githubusercontent.com/53793601/210229634-0b96daa4-6830-4631-abea-05baab94614e.JPG)

**Reply by Misterkiwi500** [reply]
Oh! wait.. it works for me! but what the error shows would, I don't understand. :/

**Reply by Misterkiwi500** [reply]
as I said, I don't understand what the error shows. Still, I'm sorry to bother you.

**Reply by DominikDoom** [reply]
It means that it can't find this attribute
https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/master/modules/textual_inversion/textual_inversion.py#L62
Which should only be the case if your webui is outdated, since this attribute is obviously in the code of the current version.

**Reply by DominikDoom** [reply]
If you still have this issue, I added a fallback to it so it doesn't throw an error (still prints a notification message about it).
Of course, if that happens, you won't get the new embedding completion features.
